src/languages/markdown.js

```diff
@@ -15,8 +15,7 @@ export default function(hljs) {
     relevance: 0
   };
   const HORIZONTAL_RULE = {
-    begin: '^[-\\*]{3,}',
-    end: '$'
+    begin: '^[ ]{0,3}([-]{3,}|[\*]{3,})[ \t]*$'
```

Comment:
- Don't we also need `_`?
- Does `\s*$` work here?
- Is the `\` needed for the `*` inside the `[]`?